### ISSUE TITLE
Two-way deno bindings

### ISSUE THREAD

**Issue by eliassjogreen** [webview/webview_deno#12]
- [ ] Calling functions in the webview from deno
- [ ] Calling deno from webview

To call deno externals from the webview an invoke handler function will be needed. I tried to implement it in [src/lib.rs:111](https://github.com/eliassjogreen/deno_webview/blob/4d49c0438e2035a4f18116a7ff07375154b78a4a/src/lib.rs#L111) but to no avail. We will also need to implement a way of calling deno callbacks from rust. To do this we will most likely use a polling solution like [signal](https://github.com/denoland/deno/blob/master/cli/ops/signal.rs) or [fs events](https://github.com/denoland/deno/blob/d8711155ca20fb2907beed304557d0815ac56452/cli/ops/fs_events.rs).

<sub>Help from some rust wizard would be pretty nice :)</sub>

**Comment by treeandgrass** [comment]
how about deno and webview share v8 isolate. 

electron impleaments:

https://github.com/electron/electron/blob/06bf0d08dc3c941ccce0dfb0f9e0e146d9232605/shell/common/node_bindings.cc

**Comment by eliassjogreen** [comment]
Probably isn't possible as the isolate isn't shared with plugins ([docs.rs](https://docs.rs/deno_core/0.45.0/deno_core/)).

**Comment by jcc10** [comment]
Could I ask what is the situation where web-sockets would not be sufficient? Since the web-view front-end (IMO) should be on a different thread then the deno back-end.

**Comment by eliassjogreen** [comment]
Ideally we want to use the [webview_bind](https://github.com/webview/webview_rust/blob/dev/webview-official-sys/src/lib.rs#L32) function but due to threading issues blocking the deno thread (because the `op_webview_run` is blocking) we cannot communicate between rust and deno and therefor not have the callbacks needed for implementing binding deno functions to a webview. Web-sockets would not work unless you run the webview in a separate web-worker, and running the webview in a separate web-worker blocks the web-worker deno message polling which means you cannot interact with the webview methods like `webview.exit`. [For further reading on the threading issue.](https://ptb.discordapp.com/channels/711271828194197529/711323679522422787/736620096243499069)

**Comment by eliassjogreen** [comment]
I solved it in release 0.5.3, it's far from perfect with a pretty bad js/ts interface but that can always be improved upon. There is an example in `examples/external`.

**Comment by eliassjogreen** [comment]
And with the release of 0.5.4 the js/ts interface is pretty good: 
```ts
for await (const event of webview.iter()) {
  console.log(event)
}
```
